pkg/cache/tas_flavor_snapshot.go

```diff
@@ -806,19 +920,27 @@ func (s *TASFlavorSnapshot) sortedDomains(domains []*domain, unconstrained bool)
 		if a.state == b.state {
 			return slices.Compare(a.levelValues, b.levelValues)
 		}
+		if a.sliceState == b.sliceState {
+			return cmp.Compare(a.state, b.state)
+		}
```

Comment:
Should sliceState be compared first, as in the end this is what we are trying to fit (until we get to levels where sliceState=0)? It is possible that a.state == b.state and a.sliceState != b.sliteState, right?

Reply:
Nice catch! I'm fixing that.